### PR TITLE
Allow prediction of probabilities and selection of scoring method.

### PR DESCRIPTION
**Pull request by pfsq**
Hi Sebastian,

I've started to use your extensions to assess my models before making submissions to Kaggle and found it quite useful so far. But in order to match Kaggle's evaluation methods for different competitions I added some functionality to the learning_curves function.

Basically, I added `predict_proba` to predict probabilities and `scoring` to select the metrics.

Regards.

### CODE REVIEW

**Comment by rasbt** [comment]
Hi, Pablo,

thanks a lot for adding those features! I made a few changes though. I didn't want to make this function 100% dependent on scikit-learn thus the sklearn.metrics is not imported by default, only if a different metric than "missclassification error" is chosen. 
Somehow I couldn't get the `predict_proba` to work and removed it for now. Unfortunately, I don't have much time for tinkering right now due to my work and book writing in my free time. But it would be nice if you could add the `predict_proba` with an example in `mlxtend/tests/tests_evaluate/test_learning_curves.py` and/or IPython notebook at `docs/examples/evaluate_plot_learning_curves.ipynb`

Anyway, good luck for the competition!

Cheers,
Sebastian
